src/Core/Echo.Ast/Construction/BlockTransformer.cs

```diff
@@ -0,0 +1,298 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Echo.ControlFlow.Blocks;
+using Echo.Core.Code;
+using Echo.DataFlow;
+
+namespace Echo.Ast.Construction
+{
+    internal sealed class BlockTransformer<TInstruction>
```

Comment:
This has too much of a generic name for a class that lifts blocks containing low level instructions to blocks with a higher level AST representation.